jetty-test-helper/src/main/java/org/eclipse/jetty/toolchain/test/jupiter/WorkDirExtension.java

```diff
@@ -113,35 +107,38 @@ private void assertValidFieldCandidate(Field field)
 
     private Path toPath(Class<?> classContext, ExtensionContext context) throws IOException
     {
-        StringBuilder dirName = new StringBuilder();
+        StringBuilder dirNameBuilder = new StringBuilder();
 
         Class<?> clazz = context.getTestClass().orElse(classContext);
-        dirName.append(StringMangler.condensePackageString(clazz.getName()));
-        dirName.append(File.separatorChar);
+        dirNameBuilder.append(StringMangler.condensePackageString(clazz.getName()));
+        dirNameBuilder.append(File.separatorChar);
 
         if (context.getTestMethod().isPresent())
         {
             String methodname = context.getTestMethod().get().getName();
             if (OS.WINDOWS.isCurrentOs())
             {
-                dirName.append(StringMangler.maxStringLength(30, methodname));
+                dirNameBuilder.append(StringMangler.maxStringLength(30, methodname));
             }
             else
             {
-                dirName.append(methodname);
+                dirNameBuilder.append(methodname);
             }
 
             if (!context.getDisplayName().startsWith(methodname))
             {
-                dirName.append(URLEncoder.encode(context.getDisplayName().trim(), UTF_8.toString()));
+                dirNameBuilder.append(URLEncoder.encode(context.getDisplayName().trim(), UTF_8.toString()));
             }
         }
         else
         {
-            dirName.append(URLEncoder.encode(context.getDisplayName().trim(), UTF_8.toString()));
+            dirNameBuilder.append(URLEncoder.encode(context.getDisplayName().trim(), UTF_8.toString()));
         }
 
-        return MavenTestingUtils.getTargetTestingPath().resolve(dirName.toString());
+        String dirName = OS.WINDOWS.isCurrentOs() ?
+                dirNameBuilder.toString().replaceAll("[^a-zA-Z0-9-]", "_") :
```

Comment:
Too aggressive, you'll wind up with duplicate / reused directories in various parameterized paths.

It would be better to URLEncode this path segment (for all OSs) instead of converting bad chars to `_`

Reply:
@joakime They are already being URL encoded. It looks like `java.net.URLEncoder` does not encode the `*` character which is causing the problems, unlike `org.eclipse.jetty.util.UrlEncoded` which will encode the `*` character.

I guess we can't have `jetty-util` dependency here, so is it okay to do `replaceAll("\\*", "%2A")`?

Reply:
Lets make sure it covers the odd cases.

We should make sure the following characters are encoded (at least), with test cases?

```
  / (slash)
  < (less than)
  > (greater than)
  : (colon - sometimes works, but is actually NTFS Alternate Data Streams)
  " (double quote)
  / (forward slash)
  \ (backslash)
  | (vertical bar or pipe)
  ? (question mark)
  * (asterisk)
  00-32 (control characters)
  127 (DEL)
```